lib/pos_transaction_register.go

```diff
@@ -161,13 +168,66 @@ func (tr *TransactionRegister) RemoveTransaction(txn *MempoolTx) error {
 	return nil
 }
 
+func (tr *TransactionRegister) addBucket(bucket *FeeTimeBucket) {
+	if bucket == nil {
```

Comment:
do we need to hold the lock for add/remove bucket? I think we only use it in contexts where we already hold a write lock, but we should name it such that it's clear that it does not hold a lock